decidim-conferences/app/commands/decidim/conferences/admin/invite_user_to_join_conference.rb

```diff
@@ -98,7 +98,7 @@ def user
 
         def invite_user_to_sign_up
           user.skip_reconfirmation!
-          user.invite!(invited_by, invitation_instructions: "join_conference", conference:)
+          user.invite!(invited_by, invitation_instructions: "join_conference", conference:, registration_type: form.registration_type)
```

Comment:
Hi @ahukkanen, I try this fix on 0.25.2 and 0.26.2 and there was an error because it misses conference after the " : "

It works like that:
user.invite!(invited_by, invitation_instructions: "join_conference", conference: conference, registration_type: form.registration_type)

Now the problem is that the links to Decline an invitation isn't displayed anymore. We just have the link to Accept the invitation:
![image](https://user-images.githubusercontent.com/105821969/186763057-a87844d1-0a53-44e3-85ff-cf0ecb7f5097.png)

Reply:
The syntax issue is due to Ruby 3.1 update (#9449). The current code syntax is correct under Ruby 3.1, otherwise the specs would fail.

Regarding the "Decline" link, the email that is being sent for new users is not the same email. New users get sent the "invitation to the platform" email which does not implement the "Decline" link.

This is how it is designed to work based on the current code:
- If the participant is an existing participant, send them the conference invitation email
- If the participant is a new participant, send them the invitation email to join the platform

After the new user joins the platform through this link, they will be automatically redirected to accept the conference invitation after a successful registration.

If they don't want to accept the invitation, they should just delete their email and nothing happens after that.

If you would like to change the current functionality, you should post a new feature proposal at MetaDecidim. I am just fixing the bug in this PR, not changing how the process is designed to work.

Reply:
@ahukkanen Thank you for the explanation, it make sense.

The thing that got me confused in the process is that **we receive the same email text for existing user and not existing user, saying you can accept or decline the invitation**:
![image](https://user-images.githubusercontent.com/105821969/186979242-af15e45b-6e28-4f35-89ae-196154e48a39.png)

Should I post a new feature proposal to make it less confusing with 2 different texts or 2 different email for existing and non existing user  or a bug fix would fix that ?

Also, i still have some issues:

- After the **non existing user Accept the invitation** and create his account, he arrives on the home page but there is an error or warning message saying "**The invitation token is not valid**":
![image](https://user-images.githubusercontent.com/105821969/186976009-3ce9adae-1cec-4f10-ad43-4128be7a7dac.png)

- After an **existing user click on Decline the invitation**, he arrives on the login page, but there is an error or warning message saying "**You are not allowed to do this action**":
![image](https://user-images.githubusercontent.com/105821969/186977048-54d5162c-c12a-44f0-b554-62018fe87912.png)

Then when the **existing user click on Decline** log in there is an error or warning message saying "**You are already connected**":
![image](https://user-images.githubusercontent.com/105821969/186977245-c0b7b0c1-4943-40a6-8010-0b2d4b03462a.png)

- After an **existing user click on Accept the invitation**, he arrives on the login page, but there is an error or warning message saying "**You are not allowed to do this action**":
![image](https://user-images.githubusercontent.com/105821969/186977874-f91073b7-68c1-473d-acad-b11c371b5d1b.png)

Then there is an error saying **The validation failed, User is not available** :
![image](https://user-images.githubusercontent.com/105821969/186978291-eaa6437c-b5c2-4d02-88ff-3aefd387f72f.png)

With this trace:
![image](https://user-images.githubusercontent.com/105821969/186978434-3a58fdf9-32b9-46e3-98e4-82c04cd368cf.png)

Should I create another bug report for that or just having the comments here is fine ?

One more thing, on try.decidim.com it is the 0.27 version runing ? How to know what version running on try.decidim.com ?

Thank you very much.

Lawrence

Reply:
Thanks for the through testing of the changes @lawrencemcn, it is greatly appreciated!

To answer your questions:

> The thing that got me confused in the process is that we receive the same email text for existing user and not existing user, saying you can accept or decline the invitation:
> 
> Should I post a new feature proposal to make it less confusing with 2 different texts or 2 different email for existing and non existing user or the bug fix would fix that ?

I have clarified the message at 196762b124fbe0057d1f6c157a6d94431b8f1696 and remove the "decline" part of the intro text as it is already mentioned below. Here is what the email looks after this change:

![Conference invitation for new participant](https://user-images.githubusercontent.com/864340/187212380-f58122c4-bb79-4cdf-b5d8-38fce07d32ec.png)

No need to post new feature proposal for that.

> After the non existing user Accept the invitation and create his account, he arrives on the home page but there is an error or warning message saying "The invitation token is not valid":

This issue is now fixed by f4563ad78c700c1757c63c884251ce3eaba664cf.

> After an existing user click on Decline the invitation, he arrives on the login page, but there is an error or warning message saying "You are not allowed to do this action":

This is by design as we need the existing user to sign in before accepting or declining the invitation with the current functionality. However, I have clarified the error messages slightly at f8382ec600447cb31b75ca00a4d78e8ee6c6cbac which might help a bit.

> Then when the existing user click on Decline log in there is an error or warning message saying "You are already connected":

This issue is now fixed by f4563ad78c700c1757c63c884251ce3eaba664cf.

> After an existing user click on Accept the invitation, he arrives on the login page, but there is an error or warning message saying "You are not allowed to do this action":

This is by design as we need the existing user to sign in before accepting or declining the invitation with the current functionality. However, I have clarified the error messages slightly at f8382ec600447cb31b75ca00a4d78e8ee6c6cbac which might help a bit.

> Then there is an error saying The validation failed, User is not available :

This issue is now fixed by f4563ad78c700c1757c63c884251ce3eaba664cf.

> Should I create another bug report for that or just having the comments here is fine ?

Here is fine as long as the issues are directly related to this PR. If it's not clearly related to the original issues or this PR, please create new bug reports if it's a bug. If you want to change how it works, please create new feature proposals at MetaDecidim.

> One more thing, on try.decidim.com it is the 0.27 version runing ? How to know what version running on try.decidim.com ?

It seems to be running 0.26.1 right now. You can check this through the technical API but I don't think there's any other way to find that out.

Reply:
Hi @ahukkanen, so with the letter opener and mock email it works fine but **with the Delayed_job and real email address  there is still an error just for the Decline link, the Server name is missing in the link**.

Here is the Decline link sent to a Gmail address :
[https://www.google.com/url?q=http:///conferences/testtest/registration/8/conference_registration/decline_invitation&source=gmail&ust=1659035444611000&usg=AOvVaw2CKPi6ACS2NpxT08lI_rUo&rct=i](https://www.google.com/url?q=http%3A///conferences/testtest/registration/8/conference_registration/decline_invitation&source=gmail&ust=1659035444611000&usg=AOvVaw2CKPi6ACS2NpxT08lI_rUo&rct=i)

Here is the Decline link sent to a Outlook address. The link is not even clickable:

![image](https://user-images.githubusercontent.com/105821969/187459233-af7f5c75-c534-4451-a347-bd165a9aab9d.png)

Reply:
Great catch once again @lawrencemcn !

This latest issue should be fixed by f319de0662bfb234e2abb739b95b97fa9d1c1c48.